.github/workflows/dependencies.yml

```diff
@@ -0,0 +1,137 @@
+    steps:
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Fetch state and labels of PR
+          # Or exit successfully if PR does not exist
+          JSON=$(gh pr view cargo_update --repo $GITHUB_REPOSITORY --json labels,state || exit 0)
+          STATE=$(echo "$JSON" | jq -r '.state')
+          WAITING_ON_BORS=$(echo "$JSON" | jq '.labels[] | any(.name == "S-waiting-on-bors"; .)')
+
+          # Exit with error if open and S-waiting-on-bors
+          if [[ "$STATE" == "OPEN" && "$WAITING_ON_BORS" == "true" ]]; then
+            exit 1
+          fi
+
+  update:
+    name: update dependencies
+    needs: not-waiting-on-bors
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout the source code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: install the bootstrap toolchain
+        run: |
+          # Extract the stage0 version
+          TOOLCHAIN=$(jq -r '.compiler | {version,date} | join("-")' -- src/stage0.json)
+          # Install and set as default
+          rustup toolchain install --no-self-update --profile minimal $TOOLCHAIN
+          rustup default $TOOLCHAIN
+
+      - name: cargo update
+        # Remove first line that always just says "Updating crates.io index"
+        run: cargo update 2>&1 | sed '/crates.io index/d' | tee -a cargo_update.log
+      - name: upload Cargo.lock artifact for use in PR
+        uses: actions/upload-artifact@v3
+        with:
+          name: Cargo-lock
+          path: Cargo.lock
+          retention-days: 1
+      - name: upload cargo-update log artifact for use in PR
+        uses: actions/upload-artifact@v3
+        with:
+          name: cargo-updates
+          path: cargo_update.log
+          retention-days: 1
+
+  pr:
+    name: amend PR
+    needs: update
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: checkout the source code
+        uses: actions/checkout@v3
+
+      - name: download Cargo.lock from update job
+        uses: actions/download-artifact@v3
+        with:
+          name: Cargo-lock
+      - name: download cargo-update log from update job
+        uses: actions/download-artifact@v3
+        with:
+          name: cargo-updates
+
+      - name: craft PR body and commit message
+        run: |
+          echo "${COMMIT_MESSAGE}" > commit.txt
+          cat cargo_update.log >> commit.txt
+
+          echo "${PR_MESSAGE}" > body.md
+          echo '```txt' >> body.md
+          cat cargo_update.log >> body.md
+          echo '```' >> body.md
+
+      - name: commit
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git switch -C cargo_update
+          git add ./Cargo.lock
+          git commit --no-verify --file=commit.txt
+
+      - name: push
+        run: git push --no-verify -f -u origin cargo_update
```

Comment:
I think this step needs to be moved after checking for existing PRs -- my thinking is that we should refuse to *update* a PR. We might be able to alter that to something like "no updates if labeled S-waiting-on-bors", but that might be too complex.

Reply:
> I think this step needs to be moved after checking for existing PRs -- my thinking is that we should refuse to *update* a PR. We might be able to alter that to something like "no updates if labeled S-waiting-on-bors", but that might be too complex.

If you don't want to update a PR at all, that's fine. But if your concern is just bors:

- The whole thing is skipped when the existing PR is open and marked S-waiting-on-bors
- When the existing PR is closed, pushes don't update the existing PR, so we push and open a new PR
- Only when the existing PR is open and not S-waiting-on-bors, push and edit that PR

That said, moving the push after the edit or open would make this less reliant on that "pushing to a closed GitHub PR doesn't update it" behavior.

Reply:
Ah, I missed that block earlier in the file -- happy to merge with or without that change. I think pushing to a closed PR makes it impossible to re-open it, so I think making this change is probably good, but non-critical. (We almost certainly will rarely or never be closing these PRs).

Reply:
Okay wasn't able to get push-after-open to work, but if that's alright with you, we should be good to go.